.golangci.yml

```diff
@@ -0,0 +1,45 @@
+run:
+  go: '1.17'
```

Comment:
Good point.  I will set it to use 1.18.  I will also clean up the _.golangci.yml_, which has things that are Jaeger specific and aren't needed for jaeger-operator.

Note that Jaeger itself uses 1.17 here.  When I set this to 1.18 I get two messages:
```
WARN [linters context] structcheck is disabled because of go1.18. You can track the evolution of the go1.18 support by following the https://github.com/golangci/golangci-lint/issues/2649. 
WARN [linters context] contextcheck is disabled because of go1.18. You can track the evolution of the go1.18 support by following the https://github.com/golangci/golangci-lint/issues/2649. 
```

We don't need those linters, but perhaps that is why Jaeger itself did not bump to Go version used for linting.

Reply:
I see. We can enable right now just the very minimum and enable other stuff later, fixing the reported warnings